libgumath/kernels/cuda_device_binary.cu

```diff
@@ -984,6 +984,12 @@ CUDA_DEVICE_ALL_COMPARISON(greater_equal, greater_equal, __hge, lexorder_ge)
 #define greater(x, y) x > y
 CUDA_DEVICE_ALL_COMPARISON(greater, greater, __hgt, lexorder_gt)
 
+#define equal(x, y) x == y
+CUDA_DEVICE_ALL_COMPARISON(equal, equal, __heq, lexorder_eq)
```

Comment:
I think just using `equal` instead of `lexorder_eq` should work. The `isnan` part of the lexorder functions should be handled correctly by `==` and `!=`.

Otherwise there's no lexorder.